test/registry/finalizeExit.js

```diff
@@ -0,0 +1,245 @@
+    });
+
+    it('should allow a listing to exit when no challenge exists', async () => {
+      const listing = utils.getListingHash('google.com');
+
+      const initialApplicantTokenHoldings = await token.balanceOf.call(applicant);
+
+      await utils.addToWhitelist(listing, paramConfig.minDeposit, applicant, registry);
+
+      const isWhitelisted = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelisted, true, 'the listing was not added to the registry');
+
+      await registry.initExit(listing, { from: applicant });
+
+      await utils.increaseTime(paramConfig.exitTimeDelay + 1);
+      await registry.finalizeExit(listing, { from: applicant });
+
+      const isWhitelistedAfterExit = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelistedAfterExit, false, 'the listing was not removed on exit');
+
+      const finalApplicantTokenHoldings = await token.balanceOf.call(applicant);
+      assert.strictEqual(
+        initialApplicantTokenHoldings.toString(10),
+        finalApplicantTokenHoldings.toString(10),
+        'the applicant\'s tokens were not returned to them after exiting the registry',
+      );
+
+      const listingStruct = await registry.listings.call(listing);
+      assert.strictEqual(listingStruct[5].toString(), '0', 'exit time did not reset');
+    });
+
+    it('should not allow a listing to finalize exit when exit was not initialized', async () => {
+      const listing = utils.getListingHash('youtube.com');
+
+      const initialApplicantTokenHoldings = await token.balanceOf.call(applicant);
+
+      await utils.addToWhitelist(listing, paramConfig.minDeposit, applicant, registry);
+
+      const isWhitelisted = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelisted, true, 'the listing was not added to the registry');
+      try {
+        await registry.finalizeExit(listing, { from: applicant });
+        assert(false, 'exit succeeded when it should have failed due to exit not being initialized');
+      } catch (err) {
+        const errMsg = err.toString();
+        assert(utils.isEVMException(err), errMsg);
+      }
+
+      const isWhitelistedAfterExit = await registry.isWhitelisted.call(listing);
+      assert.strictEqual(isWhitelistedAfterExit, true, 'the listing was removed on finalizeExit');
+
+      const finalApplicantTokenHoldings = await token.balanceOf.call(applicant);
+      assert.strictEqual(
+        finalApplicantTokenHoldings.toString(),
+        initialApplicantTokenHoldings.sub(paramConfig.minDeposit).toString(),
+        'the applicant\'s tokens were returned in spite of failing to exit',
+      );
+
+      const listingStruct = await registry.listings.call(listing);
+      assert.strictEqual(listingStruct[5].toString(), '0', 'exit time was initialized');
+    });
+
+    it('should not allow a listing to finalize exit when time is not up', async () => {
```

Comment:
I can't tell what is being tested here (what does "time is not up" mean?). Is this testing that you can't finalize the exit during the delay before the exit period starts?